tests/openassetio-core-c/private/errorsTest.cpp

```diff
@@ -33,9 +35,27 @@ SCENARIO("throwIfError error code/message handling") {
 
     WHEN("throwIfError is called") {
       THEN("expected exception is thrown") {
-        REQUIRE_THROWS_MATCHES(openassetio::throwIfError(code, cmessage), std::runtime_error,
+        REQUIRE_THROWS_MATCHES(throwIfError(code, cmessage), std::runtime_error,
                                Catch::Message("123: some error"));
       }
     }
   }
 }
+
+SCENARIO("Using extractExceptionMessage to copy a C++ exception message to a C StringView") {
+  using openassetio::errors::extractExceptionMessage;
+
+  GIVEN("An exception and a StringView") {
+    const openassetio::Str expectedMessage = "some error";
+    const std::runtime_error runtimeError{expectedMessage};
+
+    openassetio::Str storage(expectedMessage.size(), '\0');
+    OPENASSETIO_NS(StringView) actualMessage{storage.capacity(), storage.data(), 0};
+
+    WHEN("extractExceptionMessage copies the message from the exception to the StringView") {
+      extractExceptionMessage(&actualMessage, runtimeError);
+
+      THEN("StringView matches expected message") { CHECK(actualMessage == expectedMessage); }
```

Comment:
[fix] Should we also check that the copy is detached, so when the exception goes out of scope...

Reply:
Updated tests to ensure data was copied not just pointed to in d4609176